x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBytes.java

```diff
@@ -179,16 +180,26 @@ public static void copyToCacheFileAligned(
         int relativePos,
         int length,
         IntConsumer progressUpdater,
-        ByteBuffer buf
+        ByteBuffer buf,
+        CacheCopyMetricsConsumer metricsConsumer
     ) throws IOException {
+        final long copyStartTime = System.nanoTime();
         int bytesCopied = 0;
         long remaining = length;
+        int totalBytesRead = 0;
+        long totalReadTimeNanos = 0L;
+        long totalWriteTimeNanos = 0L;
```

Comment:
I think we can save changes for this method variant for now. It is not used by stateless, the code block around it is disabled in stateless. It is used by frozen teir which is not of interest at this time. If we change it, we'll have to test it and that's a bit anonying since it is not used. I'd just add a TODO here and maybe create an issue for it.

Reply:
Actually it is used in [`SearchDirectory`](https://github.com/elastic/elasticsearch-serverless/blob/15c169722f978334fdb91290f99436810c3b1f10/modules/stateless/src/main/java/co/elastic/elasticsearch/stateless/lucene/SearchDirectory.java#L375-L383) I just found out when I removed it 😞

Reply:
If you trace the method further up, you can see `downloadCommit` is disable [here](https://github.com/elastic/elasticsearch-serverless/blob/15c169722f978334fdb91290f99436810c3b1f10/modules/stateless/src/main/java/co/elastic/elasticsearch/stateless/engine/SearchEngine.java#L315)

Reply:
Removed, replaced with a TODO